File: android/src/main/kotlin/com/giphy/giphy_flutter_sdk/dto/GiphyFlutterResizeMode.kt
package com.giphy.giphy_flutter_sdk.dto

enum class GiphyFlutterResizeMode {
    CENTER,
    CONTAIN,
    COVER,
    STRETCH;

    companion object {
        val DEFAULT_MODE = COVER

        fun fromString(value: String?): GiphyFlutterResizeMode? {
            return when (value) {
                "center" -> CENTER
                "contain" -> CONTAIN
                "cover" -> COVER
                "stretch" -> STRETCH
                else -> null
            }
        }
    }
}

File: ios/Classes/dto/GiphyFlutterResizeMode.swift
//
//  GiphyFlutterResizeMode.swift
//  giphy_flutter_sdk
//
//  Created by Alex on 24.2.2024.
//

public enum GiphyFlutterResizeMode: Int {
    case center = 1
    case contain = 2
    case cover = 3
    case stretch = 4
    
    static let defaultMode: GiphyFlutterResizeMode = .cover
    
    static func fromString(value: String?) -> GiphyFlutterResizeMode? {
        switch value {
        case "center":
            return .center
        case "contain":
            return .contain
        case "cover":
            return .cover
        case "stretch":
            return .stretch
        default:
            return nil
        }
    }
}
